list/list_resource.go

```diff
@@ -41,6 +42,16 @@ type ListResource interface {
 	List(context.Context, ListRequest, *ListResultsStream)
 }
 
+// ListResourceWithProtoSchemas is an interface type that extends ListResource to include a method
+// which allows provider developers to supply the ProtoV5 representations of resource and resource identity
+// schemas. This is necessary if list functionality is being used with a legacy resource.
```

Comment:
What about protocol v6? It is technically possible with this design to have a resource defined in `terraform-plugin-go` that would be with protocol v6.